applications/GeoMechanicsApplication/geo_mechanics_application.cpp

```diff
@@ -0,0 +1,320 @@
+    KRATOS_REGISTER_VARIABLE( PARAMETER_45 )
+    KRATOS_REGISTER_VARIABLE( PARAMETER_46 )
+    KRATOS_REGISTER_VARIABLE( PARAMETER_47 )
+    KRATOS_REGISTER_VARIABLE( PARAMETER_48 )
+    KRATOS_REGISTER_VARIABLE( PARAMETER_49 )
+    KRATOS_REGISTER_VARIABLE( PARAMETER_50 )
+
+    KRATOS_REGISTER_VARIABLE(NUMBER_OF_UMAT_STATE_VARIABLES)
+    KRATOS_REGISTER_VARIABLE(NUMBER_OF_UMAT_PARAMETERS)
+
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLES )
+
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_1 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_2 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_3 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_4 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_5 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_6 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_7 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_8 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_9 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_10 )
+
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_11 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_12 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_13 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_14 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_15 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_16 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_17 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_18 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_19 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_20 )
+
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_21 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_22 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_23 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_24 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_25 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_26 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_27 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_28 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_29 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_30 )
+
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_31 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_32 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_33 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_34 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_35 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_36 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_37 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_38 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_39 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_40 )
+
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_41 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_42 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_43 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_44 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_45 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_46 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_47 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_48 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_49 )
+    KRATOS_REGISTER_VARIABLE( STATE_VARIABLE_50 )
```

Comment:
hm those names will be very hard to understand and debug
what is the intended usecase?
Maybe there is a better solution

Reply:
PARAMETER_xxx are replaced with a vector. The state_varaible_xxx are needed only for output, if a user wants to retrieve a specific state variables. The state variables in UMATs do not have names.